testsuite/classlibrary/TestServerClientID.sc

```diff
@@ -0,0 +1,75 @@
+/*
+TestServerClientID.run;
+UnitTest.gui;
+
+Tests for clientID, maxLogins and allocator creation based on these.
+
+* clientID should be impossible to set to nonsense data,
+it should always be between 0 and maxLogins - 1.
+
+NOTE: All test cases currently have powerOfTwo numbers for maxLogins.
+This is because the server currently rounds up maxLogins to nextPowerOfTwo;
+so when the server IS NOT booted, odd numbers of maxLogins stay as they are;
+when the server IS booted, the rounded-up maxLogins come back to each client.
+If this server behavior changes to support non-powerOfTwo numbers of clients,
+the tests should also include irregular numbers for maxLogins, e.g. 1, 2, 3, 5, 8.
+
+*/
+
+TestServer_clientID : UnitTest {
+	// these while server is off
+	test_ClientIDChecks {
+		var s = Server(\testID);
+		this.assert(s.clientID == 0, "s.clientID should be 0 by default.");
+		// test checks for bad input
+		s.clientID = -1.sqrt;
+		this.assert(s.clientID == 0, "s.clientID should block non-Integers.");
+		s.clientID = -123;
+		this.assert(s.clientID == 0, "s.clientID should block negative numbers.");
+		s.options.maxLogins = 32;
+		s.clientID = s.options.maxLogins;
+		this.assert(s.clientID == 0, "s.clientID should block number >= maxLogins.");
+		s.clientID = s.options.maxLogins - 1;
+		this.assert(s.clientID == 31, "s.clientID should be settable if valid.");
```

Comment:
this is a test that might fail once #3181 is merged.

Reply:
thanks for looking this through so closely! 
no, it should never fail: 
these tests are all done with a new server that is not booted, so nothings sets it from scsynth;
and I am setting it up two lines above: 
```s.options.maxLogins = 32;```
so 31 should really always be correct here.

Reply:
ok!